flow/raster_cache.cc

```diff
@@ -175,25 +176,50 @@ std::unique_ptr<RasterCacheResult> RasterCache::RasterizeDisplayList(
 
 void RasterCache::Prepare(PrerollContext* context,
                           Layer* layer,
-                          const SkMatrix& ctm) {
-  RasterCacheKey cache_key(layer->unique_id(), RasterCacheKeyType::kLayer, ctm);
+                          const SkMatrix& ctm,
+                          RasterCacheLayerStrategy strategy) {
+  RasterCacheKey cache_key = MakeRasterCacheKeyForLayer(layer, strategy, ctm);
   Entry& entry = cache_[cache_key];
   entry.access_count++;
   entry.used_this_frame = true;
   if (!entry.image) {
-    entry.image = RasterizeLayer(context, layer, ctm, checkerboard_images_);
+    entry.image =
+        RasterizeLayer(context, layer, strategy, ctm, checkerboard_images_);
+  }
+}
+
+const RasterCacheKey RasterCache::MakeRasterCacheKeyForLayer(
+    const Layer* layer,
+    RasterCacheLayerStrategy strategy,
+    const SkMatrix& ctm) const {
+  switch (strategy) {
+    case RasterCacheLayerStrategy::kLayer:
+      return RasterCacheKey(layer->unique_id(), RasterCacheKeyType::kLayer,
+                            ctm);
+    case RasterCacheLayerStrategy::kLayerChildren:
+      auto children_layers =
+          static_cast<const ContainerLayer*>(layer)->layers();
+      std::vector<uint64_t> ids;
+      std::transform(children_layers.begin(), children_layers.end(),
+                     std::back_inserter(ids),
+                     [](std::shared_ptr<Layer>& layer) -> uint64_t {
+                       return layer->unique_id();
+                     });
+      return RasterCacheKey(RasterCacheKeyID(std::move(ids)),
+                            RasterCacheKeyType::kLayerChildren, ctm);
```

Comment:
Maybe this implementation is not correct. Looks like I should distinguish whether its children is one or more, if it is one, should use the child's `unique_id` and `RasterCacheType::kLayer` to generate RasterCacheKey

Reply:
Done. When there is only a single child, we use the child's 'unique_id' to generate the cache key. So that the entry can be reused as much as possible.